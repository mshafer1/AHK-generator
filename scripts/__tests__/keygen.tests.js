const { printExceptionAndStackTraceToConsole } = require("lzutf8")

const configured_region = require("../keygen").configured_region
const each = require("jest-each").default

const triggers = [
    {
        "func": "KEY",
        "skeyValue": "a",
        "modifiers[]": [
            "CTRL",
            "ALT"],
    },
    {
        "func": "STRING",
        "skeyValue": ";implies",
    }
]

const actions = [
    {
        "option": "Send",
        "input": "b",
    },
    {
        "option": "Replace",
        "input": "b",
    },
    {
        "option": "SendUnicodeChar",
        "input": "0x2192",
    },
    {
        "Program": "cmd",
        "Window": "ahk_exe cmd.exe",
        "option": "ActivateOrOpen",
    },
    {
        "option": "LockWorkStation",
    },
    {
        "option": "TurnMonitorsOff",
    },
    {
        "option": "OpenConfig",
    },
    {
        "option": "Custom",
        "Code": `
            Send, ^c
            Sleep 50
            Run, http://www.google.com/search?q=%clipboard%
            Return`,
    },
    {
        "Program": "http://dictionary.reference.com/",
        "Window": "dictionary.com",
        "option": "ActivateOrOpenChrome",
    }
]


function* generate_test_cases() {
    var i = 0;
    for (const trigger of triggers) {
        for (const action of actions) {
            var test_case = { 0: { ...trigger, ...action } }
            yield [JSON.stringify(test_case), test_case]
        }
    }
}


describe("when keygen", () => {
    each([...generate_test_cases()]).it("given %s, then matches snapshot", (name, parsed_data) => {
        expect(configured_region(parsed_data)).toMatchSnapshot();
    })
})
