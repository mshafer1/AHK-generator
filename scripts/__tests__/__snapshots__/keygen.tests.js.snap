// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"Program":"cmd","Window":"ahk_exe cmd.exe","option":"ActivateOrOpen"}}, then matches snapshot 1`] = `
"^!a::ActivateOrOpen(\\"ahk_exe cmd.exe\\", \\"cmd\\")

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"Program":"http://dictionary.reference.com/","Window":"dictionary.com","option":"ActivateOrOpenChrome"}}, then matches snapshot 1`] = `
"^!a::ActivateOrOpenChrome(\\"dictionary.com\\", \\"http://dictionary.reference.com/\\")

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"Custom","Code":"\\n            Send, ^c\\n            Sleep 50\\n            Run, http://www.google.com/search?q=%clipboard%\\n            Return"}}, then matches snapshot 1`] = `
"^!a::
            Send, ^c
            Sleep 50
            Run, http://www.google.com/search?q=%clipboard%
            Return

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"LockWorkStation"}}, then matches snapshot 1`] = `
"^!a::LockWorkStation()

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"OpenConfig"}}, then matches snapshot 1`] = `
"^!a::
OpenConfig()
return

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"Replace","input":"b"}}, then matches snapshot 1`] = `
"^!a::
send, b
return

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"Send","input":"b"}}, then matches snapshot 1`] = `
"^!a::send, b

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"SendUnicodeChar","input":"0x2192"}}, then matches snapshot 1`] = `
"^!a::SendUnicodeChar(0x2192)

"
`;

exports[`when keygen given {"0":{"func":"KEY","skeyValue":"a","modifiers[]":["CTRL","ALT"],"option":"TurnMonitorsOff"}}, then matches snapshot 1`] = `
"^!a::TurnMonitorsOff()

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","Program":"cmd","Window":"ahk_exe cmd.exe","option":"ActivateOrOpen"}}, then matches snapshot 1`] = `
":*c:;implies::ActivateOrOpen(\\"ahk_exe cmd.exe\\", \\"cmd\\")

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","Program":"http://dictionary.reference.com/","Window":"dictionary.com","option":"ActivateOrOpenChrome"}}, then matches snapshot 1`] = `
":*c:;implies::ActivateOrOpenChrome(\\"dictionary.com\\", \\"http://dictionary.reference.com/\\")

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"Custom","Code":"\\n            Send, ^c\\n            Sleep 50\\n            Run, http://www.google.com/search?q=%clipboard%\\n            Return"}}, then matches snapshot 1`] = `
":*c:;implies::
            Send, ^c
            Sleep 50
            Run, http://www.google.com/search?q=%clipboard%
            Return

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"LockWorkStation"}}, then matches snapshot 1`] = `
":*c:;implies::LockWorkStation()

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"OpenConfig"}}, then matches snapshot 1`] = `
":*c:;implies::
OpenConfig()
return

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"Replace","input":"b"}}, then matches snapshot 1`] = `
":*c:;implies::b

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"Send","input":"b"}}, then matches snapshot 1`] = `
":*c:;implies::send, b

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"SendUnicodeChar","input":"0x2192"}}, then matches snapshot 1`] = `
":*c:;implies::SendUnicodeChar(0x2192)

"
`;

exports[`when keygen given {"0":{"func":"STRING","skeyValue":";implies","option":"TurnMonitorsOff"}}, then matches snapshot 1`] = `
":*c:;implies::TurnMonitorsOff()

"
`;
